frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1630,17 +1613,23 @@ function ReaderHighlight:saveHighlight()
             drawer = self.view.highlight.saved_drawer,
             chapter = chapter_name,
         }
-        table.insert(self.view.highlight.saved[page], hl_item)
+        -- table.insert(self.view.highlight.saved[page], hl_item)
         local bookmark_item = self:getHighlightBookmarkItem()
         if bookmark_item then
             bookmark_item.datetime = datetime
             bookmark_item.chapter = chapter_name
-            self.ui.bookmark:addBookmark(bookmark_item)
+            bookmark_item.page_num = page
+            if self.ui.document.info.has_pages then
```

Comment:
The new trend is to use `if self.ui.paging then` instead.

Reply:
Changed. Thanks.